bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/JdbcPersistenceService.java

```diff
@@ -110,39 +116,46 @@ public void deactivate(final int reason) {
     @Override
     public String getId() {
         logger.debug("JDBC::getName: returning name 'jdbc' for queryable persistence service.");
-        return "jdbc";
+        return SERVICE_ID;
     }
 
     @Override
     public String getLabel(@Nullable Locale locale) {
-        return "JDBC";
+        return SERVICE_LABEL;
     }
 
     @Override
     public void store(Item item) {
-        store(item, null);
+        internalStore(item, null, item.getState());
     }
 
-    /**
-     * @{inheritDoc
-     */
     @Override
     public void store(Item item, @Nullable String alias) {
+        // alias is not supported
+        internalStore(item, null, item.getState());
+    }
+
+    @Override
+    public void store(Item item, ZonedDateTime date, State state) {
+        internalStore(item, date, state);
+    }
+
+    private void internalStore(Item item, @Nullable ZonedDateTime date, State state) {
         // Do not store undefined/uninitialized data
-        if (item.getState() instanceof UnDefType) {
+        if (state instanceof UnDefType) {
             logger.debug("JDBC::store: ignore Item '{}' because it is UnDefType", item.getName());
             return;
         }
         if (!checkDBAccessability()) {
             logger.warn(
-                    "JDBC::store:  No connection to database. Cannot persist item '{}'! Will retry connecting to database when error count:{} equals errReconnectThreshold:{}",
-                    item, errCnt, conf.getErrReconnectThreshold());
+                    "JDBC::store: No connection to database. Cannot persist state '{}' for item '{}'! Will retry connecting to database when error count:{} equals errReconnectThreshold:{}",
+                    state, item, errCnt, conf.getErrReconnectThreshold());
             return;
         }
         long timerStart = System.currentTimeMillis();
-        storeItemValue(item);
-        logger.debug("JDBC: Stored item '{}' as '{}' in SQL database at {} in {} ms.", item.getName(), item.getState(),
-                new java.util.Date(), System.currentTimeMillis() - timerStart);
+        storeItemValue(item, state, date);
+        logger.debug("JDBC: Stored item '{}' as '{}' in SQL database at {} in {} ms.", item.getName(), state,
```

Comment:
It would also make sense to make this non trivial logging conditional with `if (logger.isDebugEnabled()) {`